internal/app/machined/pkg/controllers/k8s/kubelet_service.go

```diff
@@ -196,6 +196,25 @@ func (ctrl *KubeletServiceController) Run(ctx context.Context, r controller.Runt
 			}
 		}
 
+		oldPolicy, err := loadPolicyFromFile("/var/lib/kubelet/cpu_manager_state")
+		if err != nil {
+			return fmt.Errorf("error loading cpu_manager_state file: %w", err)
+		}
+
+		policy, err := getFromMap[string](cfgSpec.Config, "cpuManagerPolicy")
+		if err != nil {
+			return err
+		}
+
+		newPolicy := policy.ValueOrZero()
+		if !equalPolicy(oldPolicy, newPolicy) {
+			logger.Info("cpuManagerPolicy changed", zap.String("old", oldPolicy), zap.String("new", newPolicy))
+
+			if removeErr := os.Remove("/var/lib/kubelet/cpu_manager_state"); removeErr != nil {
```

Comment:
this seems to be wrong, as the state file might not exist, while there's a policy change - e.g. starting up with a non-default policy

Reply:
Yeah, I agree - if there is no file, there is nothing we can do.